src/app/shared/services/current-content.service.ts

```diff
@@ -33,7 +33,12 @@ export interface ItemDetails {
   validated?: boolean,
 }
 
-export interface GroupInfo extends ContentInfo { type: 'group' }
+type GroupId = string;
+
+export interface GroupInfo extends ContentInfo {
+  id: GroupId,
+  type: 'group'
+}
```

Comment:
Why did you specify the type of GroupId separately instead of doind something like `id: string` ?

Reply:
To be consistent with the id of the `ItemInfo` interface (which is of the type `ItemId`, which is a `string)`

Reply:
Actually, I do that quite often. 
The reasons are that in type theory, a groupId and an attemptId are completely different types, they are not compatible between each other, there is no place you can use one at the place of the other one, the same as between a number and an array of strings. The "string" here is just the container for the actual type (as an array of bytes is a common container for number and the array of strings). So without pushing the concept to hard typing every variable, I use it for a main usage: when the variable name is "id"or used as "id" in its usage.  This way we do not use it by mistake in an arg which is a attemptId, a itemId, ...
It is not generalized to everywhere in the app for the moment to prevent importing tons of ids everywhere and it is only useful at some places, but it could be.

Reply:
Okay, thanks !